app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1382,103 +1379,36 @@ public void onDialogDismissed(int dialogId) {
-        public Dialog onCreateDialog(Bundle savedInstanceState) {
-            Bundle args = requireArguments();
-            textFieldEdit = (LoyaltyCardField) args.getSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY);
-            minDate = (Date) args.getSerializable(MIN_DATE_ARGUMENT_KEY);
-            maxDate = (Date) args.getSerializable(MAX_DATE_ARGUMENT_KEY);
-            // Use the current date as the default date in the picker
-            final Calendar c = Calendar.getInstance();
-
-            Date date = (Date) args.getSerializable(CURRENT_DATE_ARGUMENT_KEY);
-            if (date != null) {
-                c.setTime(date);
-            }
-
-            int year = c.get(Calendar.YEAR);
-            int month = c.get(Calendar.MONTH);
-            int day = c.get(Calendar.DAY_OF_MONTH);
-
-            // Create a new instance of DatePickerDialog and return it
-            DatePickerDialog datePickerDialog = new DatePickerDialog(getActivity(), this, year, month, day);
-            datePickerDialog.getDatePicker().setMinDate(minDate != null ? minDate.getTime() : getDefaultMinDateOfDatePicker());
-            datePickerDialog.getDatePicker().setMaxDate(maxDate != null ? maxDate.getTime() : getDefaultMaxDateOfDatePicker());
-            return datePickerDialog;
-        }
-
-        private long getDefaultMinDateOfDatePicker() {
-            Calendar minDateCalendar = Calendar.getInstance();
-            minDateCalendar.set(1970, 0, 1);
-            return minDateCalendar.getTimeInMillis();
-        }
-
-        private long getDefaultMaxDateOfDatePicker() {
-            Calendar maxDateCalendar = Calendar.getInstance();
-            maxDateCalendar.set(2100, 11, 31);
-            return maxDateCalendar.getTimeInMillis();
-        }
-
-        public void onDateSet(DatePicker view, int year, int month, int day) {
-            Calendar c = new GregorianCalendar();
-            c.set(Calendar.YEAR, year);
-            c.set(Calendar.MONTH, month);
-            c.set(Calendar.DAY_OF_MONTH, day);
-            c.set(Calendar.HOUR_OF_DAY, 0);
-            c.set(Calendar.MINUTE, 0);
-            c.set(Calendar.SECOND, 0);
-            c.set(Calendar.MILLISECOND, 0);
-
-            long unixTime = c.getTimeInMillis();
+    private void showDatePicker(
+            @Nullable Date minDate,
+            @Nullable Date maxDate,
+            MaterialPickerOnPositiveButtonClickListener<Long> listener
+    ) {
+        // Create a new instance of MaterialDatePicker and return it
+        CalendarConstraints calendarConstraints = new CalendarConstraints.Builder()
+                .setStart(minDate != null ? minDate.getTime() : getDefaultMinDateOfDatePicker())
+                .setEnd(maxDate != null ? maxDate.getTime() : getDefaultMaxDateOfDatePicker())
+                .build();
+
+        MaterialDatePicker<Long> materialDatePicker = MaterialDatePicker.Builder.datePicker()
+                .setSelection(Calendar.getInstance().getTimeInMillis())
```

Comment:
This looks like it's always using the current time instead of the current value of the date field. See above.

Reply:
Done